bluemira/plasma_physics/scaling_laws.py

```diff
@@ -0,0 +1,255 @@
+
+        min_terms = np.zeros(len(self))
+        max_terms = np.zeros(len(self))
+        for i, (arg, exp, err) in enumerate(zip(args, self.exponents, self.errors)):
+            term_values = [arg ** (exp - err), arg ** (exp + err)]
+            min_terms[i] = min(term_values)
+            max_terms[i] = max(term_values)
+
+        return min(constant_range) * np.prod(min_terms), max(constant_range) * np.prod(
+            max_terms
+        )
+
+    def __len__(self):
+        """
+        Get the length of the PowerLawScaling object.
+        """
+        return len(self.exponents)
+
+
+def lambda_q(B_t: float, q_cyl: float, p_sol: float, R_0: float, error: bool = False):
+    """
+    Scrape-off layer power width scaling (Eich et al., 2011) [4]
+
+    \t:math:`\\lambda_q=(0.73\\pm0.38)B_t^{-0.78\\pm0.25}q_{95}^{1.2\\pm0.27}P_{SOL}^{0.1\\pm0.11}R_{0}^{0.02\\pm0.2}`
+
+    Parameters
+    ----------
+    B_t: float
+        Toroidal field [T]
+    q_cyl: float
+        Cylindrical safety factor
+    p_sol: float
+        Power in the scrape-off layer [W]
+    R_0: float
+        Major radius [m]
+    method: str
+        Scaling to use when calculating lambda_q
+    error: bool
+        Whether or not to report the value with +/- errors
+
+    Returns
+    -------
+    lambda_q: float
+        Scrape-off layer width at the outboard midplane [m]
+
+    Notes
+    -----
+    [4] Eich et al., 2011
+        <https://journals.aps.org/prl/pdf/10.1103/PhysRevLett.107.215001>
+    """  # noqa: W505
+    law = PowerLawScaling(
+        constant=0.73e-3,
+        constant_err=0.38e-3,
+        exponents=[-0.78, 1.2, 0.1, 0.02],
+        exp_errs=[0.25, 0.27, 0.11, 0.20],
+    )
+    p_sol /= 1e6
+    value = law(B_t, q_cyl, p_sol, R_0)
+    if error:
+        min_value, max_value = law.calculate_range(B_t, q_cyl, p_sol, R_0)
+        return value, min_value, max_value
+    else:
+        return value
+
+
+def P_LH(n_e, B_t, A, R_0, error=False):  # noqa: N802
+    """
+    Power requirement for accessing H-mode, Martin scaling [3]
+
+    Parameters
+    ----------
+    n_e: float
+        Electron density [1/m^3]
+    B_t: float
+        Toroidal field at the major radius [T]
+    A: float
+        Plasma aspect ratio
+    R_0: float
+        Plasma major radius [m]
+
+    Returns
+    -------
+    P_LH: float
+        Power required to access H-mode [W]
+
+    Notes
+    -----
+    [3] Martin et al., 2008,
+    <https://infoscience.epfl.ch/record/135655/files/1742-6596_123_1_012033.pdf>
+    equation (3)
+
+    \t:math:`P_{LH}=2.15e^{\\pm 0.107}n_{e20}^{0.782 \\pm 0.037}`
+    \t:math:`B_{T}^{0.772 \\pm 0.031}a^{0.975 \\pm 0.08}R_{0}^{0.999 \\pm 0.101}`
+    """  # noqa :W505
+    law = PowerLawScaling(
+        constant=2.15e6,
+        constant_err=0.0,
+        exponents=[0, 0.782, 0.772, 0.975, 0.999],
+        exp_errs=[0.107, 0.037, 0.031, 0.08, 0.101],
+    )
+    leading_term = np.exp(1)
+    n_e20 = n_e / 1e20
```

Comment:
possibly overkill but:
```suggestion
    n_e20 = raw_uc(n_e,  '1/m^3', '1e20/m^3')
```